### PR TITLE
Pass through x-ms-srs header from android broker to OneAuth/MSAL

### PR DESCRIPTION
**Pull request by chase-hawthorne**
The purpose of this PR is pass the ESTS header x-ms-srs through to OneAuth/MSAL so that the MATS team can create metrics to measure the impact and reliability of CCS during outages.

### CODE REVIEW

File: common4j/src/main/com/microsoft/identity/common/java/providers/microsoft/microsoftsts/MicrosoftStsOAuth2Strategy.java [hunk 1]
```diff
@@ -644,6 +645,24 @@ protected TokenResult getTokenResultFromHttpResponse(@NonNull final HttpResponse
                 SpanExtension.current().setAttribute(
                         AttributeName.ccs_request_id.name(), ccsRequestId);
             }
+
+            final String ccsRequestSequence = response.getHeaderValue(XMS_CCS_REQUEST_SEQUENCE, 0);
+            if (null != ccsRequestSequence){
```
Comment (melissaahn) [inline common4j/src/main/com/microsoft/identity/common/java/providers/microsoft/microsoftsts/MicrosoftStsOAuth2Strategy.java:644]: Lines 649 - 665 look extremely similar to lines 631 - 647. Could some refactoring be done? For example, I think null checking tokenResponse and creating that mapWithAdditionalEntries could be done only once.

**Review by melissaahn — APPROVED** [review]
Have one comment about refactoring one section.

File: common4j/src/main/com/microsoft/identity/common/java/providers/microsoft/microsoftsts/MicrosoftStsOAuth2Strategy.java (hunk 1 above)
Comment (shahzaibj) [reply common4j/src/main/com/microsoft/identity/common/java/providers/microsoft/microsoftsts/MicrosoftStsOAuth2Strategy.java:644]: +1 on this

**Review by shahzaibj — APPROVED** [review]
Approved with suggestions

File: common4j/src/main/com/microsoft/identity/common/java/providers/microsoft/microsoftsts/MicrosoftStsOAuth2Strategy.java (hunk 1 above)
Comment (chase-hawthorne) [reply common4j/src/main/com/microsoft/identity/common/java/providers/microsoft/microsoftsts/MicrosoftStsOAuth2Strategy.java:644]: Refactored to include above suggestions

**Pull request merged**